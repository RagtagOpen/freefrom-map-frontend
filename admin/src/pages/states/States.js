import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

// Redux
import { getStateData, selectStates } from 'store/stateSlice';

// Components
import StateGrade from "./StateGrade";

function States() {
    const state = useSelector(selectStates);
    const dispatch = useDispatch();

    useEffect(() => {
        if (state.loaded === false) {
            dispatch(getStateData());
        }
    });

    function renderStates(state, index) {
        return (
            <div className="card" key={`state-${index}`}>
                <div className="card-header" id={`heading-${index}`}>
                    <h5 className="mb-0 d-flex justify-content-between align-items-center">
                        <div className="align-items-start">
                            <span style={{width: "200px", display: "inline-block"}}>{ state.name }</span>
                            <StateGrade grade={ state.grade } />
                        </div>
                        <a className="btn btn-sm btn-primary ml-4" href={`/states/${state.abbrev}`}>Edit</a>
                    </h5>
                </div>
            </div>
        )
    }

    return (
        <div>
            <h1>State Scores</h1>
            <div id="accordion">
                { state.data.map((state, index) => renderStates(state, index)) }
            </div>
            <hr className="pb-0" />
            <div className="d-flex align-items-end justify-content-end">
                <a href="/states/new" title="Create new state" className="btn btn-primary">Create</a>
            </div>
        </div>
    );
}

export default States;
