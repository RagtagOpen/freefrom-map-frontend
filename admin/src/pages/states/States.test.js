import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import States from './States';

let mockState = require('mock/store_mock.json');
const mockStore = configureStore([]);

describe('States Component', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore(mockState);

        component = renderer.create(
            <Provider store={ store }>
                <States />
            </Provider>
        );
    });

    it('should match snapshot', () => {
        expect(component.toJSON()).toMatchSnapshot();
    })
})
