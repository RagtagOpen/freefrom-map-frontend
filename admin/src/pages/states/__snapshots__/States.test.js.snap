// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`States Component should match snapshot 1`] = `
<div>
  <h1>
    State Scores
  </h1>
  <div
    id="accordion"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
        id="heading-0"
      >
        <h5
          className="mb-0 d-flex justify-content-between align-items-center"
        >
          <div
            className="align-items-start"
          >
            <span
              style={
                Object {
                  "display": "inline-block",
                  "width": "200px",
                }
              }
            >
              State Name
            </span>
            <div
              className="btn undefined"
            />
          </div>
          <a
            className="btn btn-sm btn-primary ml-4"
            href="/states/SN"
          >
            Edit
          </a>
        </h5>
      </div>
    </div>
  </div>
  <hr
    className="pb-0"
  />
  <div
    className="d-flex align-items-end justify-content-end"
  >
    <a
      className="btn btn-primary"
      href="/states/new"
      title="Create new state"
    >
      Create
    </a>
  </div>
</div>
`;
